main.c

```diff
@@ -229,8 +189,62 @@ on_check_config(TCMUService1 *interface,
 	return TRUE;
 }
 
+static gboolean
+on_change_medium(TCMUService1 *interface,
+		 GDBusMethodInvocation *invocation,
+		 gchar *name,
+		 guint64 size,
+		 gchar *cfgstring,
+		 gpointer user_data)
+{
+	struct tcmulib_context *ctx = user_data;
+	struct tcmur_handler *rhandler;
+	struct tcmu_device *dev = NULL;
+	char *reason = NULL;
+	int ret = 0;
+
+	dev = tcmulib_lookup_dev_by_tcmu_name(ctx, name);
+	if (!dev) {
+		ret = ENODEV;
+		reason = "Device not found.";
+		goto exit;
+	}
+
+	rhandler = tcmu_get_runner_handler(dev);
+	if (!rhandler->medium_change_supp) {
+		ret = EOPNOTSUPP;
+		reason = "Handler does not support medium changes.";
+		goto exit;
+	}
+
```

Comment:
Shoot. Giving myself a review comment.

We need to add lock or refcounting around parts of this to prevent the race where this is running along side a device removal.